File: src/client/finance_planner/lib/Account.dart

class Account {
  final String name;
  final String id;
  final String currency;
  final double balance;

  const Account(
      {required this.name,
      required this.id,
      required this.currency,
      required this.balance});

  factory Account.fromJson(Map<String, dynamic> json) {
    return Account(
      name: json['name'],
      id: json['id'],
      currency: json['currency'],
      balance: json['balance'],
    );
  }

  static listFromJson(List<dynamic> json) {
    var accounts = <Account>[];

    for (var element in json) {
      accounts.add(Account(
        name: element['name'],
        id: element['id'],
        currency: element['currency'],
        balance: element['balance'],
      ));
    }

    return accounts;
  }
}

File: src/backend/services/user-assets/src/main/kotlin/com/gmcn/userassets/model/Account.kt
package com.gmcn.userassets.model

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.util.*

@Document(collection = "accounts")
data class Account(
    var name: String,
    var currency: String,
    @JsonIgnore
    var userId: String,
    var balance: Float = 0.0f
) : OwnedModel(userId) {
    @Id
    @GeneratedValue
    lateinit var id: String

    private var internalReference: String = UUID.randomUUID().toString()

    @JsonProperty
    fun setInternalReference(value: String?) {
        internalReference = value ?: UUID.randomUUID().toString()
    }
}